jvm-packages/xgboost4j-spark/src/test/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostConfigureSuite.scala

```diff
@@ -29,15 +29,19 @@ class XGBoostConfigureSuite extends FunSuite with PerTest {
       .config("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
       .config("spark.kryo.classesToRegister", classOf[Booster].getName)
 
-  test("nthread configuration must be no larger than spark.task.cpus") {
-    val training = buildDataFrame(Classification.train)
-    val paramMap = Map("eta" -> "1", "max_depth" -> "2", "verbosity" -> "1",
-      "objective" -> "binary:logistic", "num_workers" -> numWorkers,
-      "nthread" -> (sc.getConf.getInt("spark.task.cpus", 1) + 1))
-    intercept[IllegalArgumentException] {
-      new XGBoostClassifier(paramMap ++ Seq("num_round" -> 2)).fit(training)
-    }
-  }
+  /*
+   * Disable this test because PR 5774 allows nthread to be larger than
```

Comment:
this is wrong. if you use more than spark.task.cpus cores in each worker, that means you silently oversubscribe cluster resources

Reply:
For this PR to work, the XGBoost parameter `nthread` needs to be the same as `spark.executor.cores`, and `spark.task.cpus` is set to 1. This means `nthread` has to be larger than `spark.task.cpus`. This PR works by having one task per executor. This should not lead to oversubscription.

Reply:
so I am seeing some assumptions here

1. users has to set spark.task.cpus to 1 to use xgb-spark

2. you assign one core to each executor process and actually fork many threads in each jvm process, and each thread are essentially compute-intensive for xgb task, what's the point to do that?

Reply:
We are assigning `spark.executor.cores` cores to each executor. The value is typically larger than 1. For example, in the 7x8 config we assign 8 cores to each of the 7 executors. With `spark.task.cpu=1` we can start 8 single-threaded tasks on an executor. They create 8 per-task DMatrices. These DMatrices are then combined into one large DMatrix and the 8 tasks stop. Next, one training task starts on the executor, which has 8 cores at its disposal. The training task can now rely on OpenMP (starting 8 OpenMP threads) to achieve good parallelism. There aren't oversubscription issues. It is just the same resources being used in different ways in different stages. This is the essence of this  PR. Thanks.